salescustomeranalysis/src/main/java/MetricsCollector.java

```diff
@@ -0,0 +1,101 @@
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Collects analytical metrics over a list of orders.
+ * The instance holds a reference to the provided orders list.
+ */
+public class MetricsCollector {
+
+    private final List<Order> orders;
+
+    /**
+     * Creates a new metrics collector over the given orders.
+     *
+     * @param orders source orders to analyze; must be non-null
+     */
+    public MetricsCollector(List<Order> orders) {
+        this.orders = orders;
+    }
+
+    /**
+     * Returns distinct customer cities found across all orders.
+     *
+     * @return list of unique cities (encounter order)
+     */
+    public List<String> getUniqueCities() {
+        return orders.stream()
+                .map(order -> order.getCustomer().getCity())
+                .distinct()
+                .toList();
+    }
+
+    /**
+     * Calculates total income from delivered orders only.
+     * Income is the sum of quantity * price over all items in delivered orders.
+     *
+     * @return total income; 0.0 if none
+     */
+    public double getTotalIncome() {
+        return orders.stream()
+                .filter(order -> order.getStatus() == OrderStatus.DELIVERED)
+                .flatMap(order -> order.getItems().stream())
+                .mapToDouble(item -> item.getQuantity() * item.getPrice())
+                .sum();
+    }
+
+    /**
+     * Finds the most frequently ordered product by name across all orders.
+     *
+     * @return product name if present; empty if there are no items
+     */
+    public Optional<String> getMostPopularProduct() {
```

Comment:
method counts product occurrences but doesn't account for quantity. A product ordered once with quantity 100 should be more popular than a product ordered twice with quantity 1